src/validation.cpp

```diff
@@ -4303,7 +4304,15 @@ bool LoadMempool(void)
                 if (state.IsValid()) {
                     ++count;
                 } else {
-                    ++failed;
+                    // mempool may contain the transaction already, e.g. from
+                    // wallet(s) having loaded it while we were processing
+                    // mempool transactions; consider these as valid, instead of
+                    // failed, but mark them as 'already existed'
+                    if (mempool.exists(tx->GetHash())) {
```

Comment:
Check `state.GetRejectCode() == REJECT_DUPLICATE`? See [this](https://github.com/bitcoin/bitcoin/blame/master/src/validation.cpp#L476).

Also, IMO this is prettier:
```cpp
if (state.IsValid()) {
    ...
} else if (state.GetRejectCode() == REJECT_DUPLICATE) {
    ...
} else {
    ...
}
```